src/components/sensor-graph.tsx

```diff
@@ -69,25 +71,42 @@ export default class SensorGraph extends React.Component<SensorGraphProps, Senso
     }
 
     yLabel() {
-        const { sensorRecording } = this.props;
+        const { sensorRecording, preRecording } = this.props;
+        const source = sensorRecording || preRecording;
         // label the data (if any) or the current sensor (if no data)
-        return sensorRecording?.name
-                ? `${sensorRecording.name} (${sensorRecording.unit})`
+        return source?.name
+                ? `${source.name} (${source.unit})`
                 : "Sensor Reading (-)";
     }
 
     renderGraph(graphWidth:number|null) {
-        const { sensorRecording } = this.props,
-              { yMin, yMax } = this.state,
-              plotYMin = yMin != null ? yMin : (sensorRecording?.min != null ? sensorRecording.min : 0),
-              plotYMax = yMax != null ? yMax : (sensorRecording?.max != null ? sensorRecording.max : 10);
+        const { sensorRecording, preRecording } = this.props;
+        const { yMin, yMax } = this.state;
+        const source = sensorRecording || preRecording;
+        const plotYMin = yMin != null ? yMin : (source?.min != null ? source.min : 0);
+        const plotYMax = yMax != null ? yMax : (source?.max != null ? source.max : 10);
+
+ let data:number[][] = [];
+        const hasSensorData = sensorRecording && sensorRecording.data.length > 0;
+        // Dygraph requires each row of data to have the same number of columns.
+        // If we have both the sensor and pre-recording data, plot both.
+        if (preRecording && hasSensorData) {
```

Comment:
The `mergeTimeSeriesData` doesn't handle empty data arrays well and this conditional doesn't check the length of the `preRecording` array.  If it has a 0 length `mergeTimeSeriesData` is going to fail.

Reply:
Fixed in [a58a392](https://github.com/concord-consortium/sensor-interactive/pull/80/commits/a58a3927a52fc291015f8ef63f01613772e190a6)